idaes/surrogate/pysmo/sampling.py

```diff
@@ -314,6 +316,100 @@ def data_sequencing(self, no_samples, prime_base):
         sequence_decimal = sequence_decimal.reshape(sequence_decimal.shape[0], )
         return sequence_decimal
 
+    def selection_columns_preprocessing(self, data_input, xlabels, ylabels):
+        """
+        Pre-processing data for multiple output selection case.
+        
+        Args:
+            data_input:     data supplied by user (dataframe or numpy array)
+            xlabels:        list of input variables
+            ylabels:        list of output variab;es
+        """
+        self.df_flag = True
+
+        if isinstance(data_input, pd.DataFrame):
+            if xlabels is None:
+                xlabels = []
+            if ylabels is None:
+                ylabels = []
+            set_of_labels = xlabels + ylabels
+
+            if not set_of_labels: 
+                # Empty list assumes one output in final column - all other variables treated as inputs
```

Comment:
Just wanted to remark on the comments here: I think they're an excellent example of what a comment should be, i.e. complement the code with useful high-level information that clarifies the purpose of the implementation.

Reply:
Thanks